oauthdoctor/diag/config.go

```diff
@@ -227,35 +227,39 @@ func (c *ConfigFile) ReplaceConfig(key, value string) string {
 	// Create a temp file
 	tmpfile, err := ioutil.TempFile("", "googleadsapi_client_lib_config")
 	if err != nil {
-		log.Fatalf("ERROR: Problem creating temp file (%s)", err.Error())
+		log.Fatalf("ERROR: Problem creating temp file.\nSystem error: %s", err)
 	}
+	defer tmpfile.Close()
 
 	// Open config file
 	configFp := filepath.Join(c.Filepath, c.Filename)
 	f, err := os.Open(configFp)
 	if err != nil {
-		log.Fatalf("ERROR: Problem opening config file (%s)", err.Error())
+		log.Fatalf("ERROR: Problem opening config file.\nSystem error: %s", err)
 	}
 	defer f.Close()
 
 	// Replace with new config value and write to temp file
 	newConfigStr := c.ReplaceConfigFromReader(key, value, f)
 	if _, err := tmpfile.Write([]byte(newConfigStr)); err != nil {
-		log.Fatalf("ERROR: Cannot write to temp config file (%s).\n%s",
-			tmpfile.Name(), err.Error())
+		log.Fatalf("ERROR: Cannot write to temp config file (%s).\nSystem error: %s",
+			tmpfile.Name(), err)
 	}
 
+	f.Close()
+	tmpfile.Close()
+
 	// Swap new config file for the old one, and backup the old file
-	backupFp := configFp + "_" + time.Now().Format(time.RFC3339)
+	backupFp := configFp + "_" + time.Now().Format("2006-01-02_15-04-05")
 	log.Printf("Backing up config file %s to %s...", configFp, backupFp)
 	if err = os.Rename(configFp, backupFp); err != nil {
-		log.Fatalf("ERROR: Cannot rename config file from (%s) to (%s).",
-			configFp, backupFp)
+		log.Fatalf("ERROR: Cannot rename config file from (%s) to (%s).\nSystem error: %s",
```

Comment:
We shouldn't categorize this as a system error. It could be misleading.

log.Fatalf("ERROR: Cannot rename config file from (%s) to (%s): ", args...) is preferable for all error statements. State what you attempted and print the error returned by Go. If you need greater granularity, you can prefix the error message with the name of the function.

Note that error messages should not end with punctuation. golint will call you an this.